modules/rte_mis_core/src/Helper/ConfigManager.php

```diff
@@ -0,0 +1,519 @@
+
+      // Add all translations.
+      foreach ($this->languageManager->getLanguages() as $language) {
+        // Do not translate for default language.
+        if ($language->isDefault()) {
+          continue;
+        }
+
+        $this->updateConfigTranslations($config_id, $language->getId(), $module_name, $path);
+      }
+    }
+  }
+
+  /**
+   * Get updated data to store in config storage.
+   *
+   * Use mode as ConfigManager::MODE_ADD_MISSING if you want to add only
+   * the newly added configuration values (defaults).
+   *
+   * @param array $existing
+   *   Existing value from config storage.
+   * @param array $data
+   *   Config data from code.
+   * @param string $mode
+   *   Mode to use replace/merge.
+   * @param array $options
+   *   Array of Keys to replace when using MODE_REPLACE_KEY.
+   *
+   * @return array
+   *   Updated data based on mode.
+   */
+  public function getUpdatedData(array $existing, array $data, $mode, array $options = []) {
+    switch ($mode) {
+      case self::MODE_ADD_MISSING:
+        // For now we check only level one keys. We may want to enhance it
+        // later to do recursive check. We may want to complicate this a bit
+        // more to handle more scenarios. For now it is simple.
+        $data = array_merge($data, $existing);
+        break;
+
+      case self::MODE_ADD_MISSING_RECURSIVE:
+        // Add Missing keys recursively, Keeping existing data as is.
+        $data = NestedArray::mergeDeepArray([$data, $existing], TRUE);
+        break;
+
+      case self::MODE_MERGE:
+        // Same as $config->merge(). To keep code consistent we do it here.
+        $data = NestedArray::mergeDeepArray([$existing, $data], TRUE);
+        break;
+
+      case self::MODE_REPLACE_KEY:
+        foreach ($options['replace_keys'] as $replace_key) {
+          $data_replace_source = $data;
+          foreach (explode('.', $replace_key) as $key) {
+            $data_replace_source = $data_replace_source[$key];
+            $data_replace_target = &$data_replace_target[$key];
+          }
+          $data_replace_target = $data_replace_source;
+        }
+        $data = $existing;
+        break;
+
+      case self::MODE_RESAVE:
+        // We just want the overrides to be applied and not actually change
+        // anything in existing config or re-read from config yaml.
+        $data = $existing;
+        break;
+
+      case self::USE_FROM_REPLACE:
+        if (!empty($options['config_name'])) {
+          foreach ($this->moduleHandler->getModuleList() as $module) {
+            $module_path = $this->fileSystem->realpath('modules/' . $module->getName());
+            $override_path = $module_path . '/config/replace/' . $options['config_name'] . '.yml';
+
+            // If there is a replace, we replace the complete configuration.
+            if (file_exists($override_path)) {
+              $data = Yaml::parse(file_get_contents($override_path));
+            }
+          }
+        }
```

Comment:
@RohanSingh-inno Can we try this? For file name update? Placing the configuration inside replace folder.